armv7a9-zynq7000/syspage.c

```diff
@@ -0,0 +1,501 @@
+			syspage_common.maps[mapID].top = syspage_common.maps[mapID].entry[i].end;
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+
+/* Initialization function */
+
+void syspage_init(void)
+{
+	int i, j;
+
+	syspage_common.argCnt = 0;
+	syspage_common.mapsCnt = 0;
+	syspage_common.progsCnt = 0;
+	syspage_common.syspage = NULL;
+
+	for (i = 0; i < MAX_ENTRIES_NB; ++i) {
+		syspage_common.entries[i].start = 0;
+		syspage_common.entries[i].end = 0;
+	}
+
+	for (i = 0; i < MAX_MAPS_NB; ++i) {
+		for (j = 0; j < MAX_ENTRIES_NB; ++j) {
+			syspage_common.maps[i].entry[j].start = 0;
+			syspage_common.maps[i].entry[j].end = 0;
+		}
+	}
+}
+
+
+
+/* Syspage's location functions */
+
+int syspage_setAddress(void *addr)
+{
+	u32 sz;
+
+	/* Assign syspage to the map */
+	syspage_common.syspage = (void *)addr;
+	sz = (u32)(MAX_SYSPAGE_SIZE / 0x200) * 0x200 + (MAX_SYSPAGE_SIZE % 0x200 ? 0x200 : 0); /* allign to PAGE_SIZE */
+
+	syspage_addEntries((u32)addr, sz);
+
+	syspage_common.syspage->arg = NULL;
+	syspage_common.syspage->maps = NULL;
+	syspage_common.syspage->progs = NULL;
+
+	syspage_common.syspage->mapssz = 0;
+	syspage_common.syspage->progssz = 0;
+	syspage_common.syspage->syspagesz = sizeof(syspage_t);
+
+	syspage_common.syspage->kernel.data = NULL;
+	syspage_common.syspage->kernel.datasz = 0;
+
+	syspage_common.syspage->kernel.bss = NULL;
+	syspage_common.syspage->kernel.bsssz = 0;
+
+	syspage_common.syspage->kernel.text = NULL;
+	syspage_common.syspage->kernel.textsz = 0;
+
+	return 0;
+}
+
+
+void *syspage_getAddress(void)
+{
+	return (void *)syspage_common.syspage;
+}
+
+
+/* General functions */
+
+void syspage_save(void)
+{
+	int i;
+
+	syspage_common.argCnt++; /* The last char is '\0' */
+	syspage_common.syspage->arg = (char *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
+	syspage_common.syspage->syspagesz += syspage_common.argCnt;
+	low_memcpy((void *)(syspage_common.syspage->arg), syspage_common.args, syspage_common.argCnt);
+
+	syspage_common.syspage->progs = (syspage_program_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
+	syspage_common.syspage->progssz = syspage_common.progsCnt;
+	syspage_common.syspage->syspagesz += (syspage_common.progsCnt * sizeof(syspage_program_t));
+	low_memcpy((void *)(syspage_common.syspage->progs), syspage_common.progs, syspage_common.progsCnt * sizeof(syspage_program_t));
+
+	syspage_common.syspage->maps = (syspage_map_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
+	for (i = 0; i < syspage_common.mapsCnt; ++i)
+		low_memcpy((void *)((void *)syspage_common.syspage->maps + i * sizeof(syspage_map_t)), &syspage_common.maps[i].map, sizeof(syspage_map_t));
+
+	syspage_common.syspage->syspagesz += (syspage_common.mapsCnt * sizeof(syspage_map_t));
+	syspage_common.syspage->mapssz = syspage_common.mapsCnt;
+}
+
+
+/* TODO: align text and show map's attributes as string based on integer */
+void syspage_show(void)
+{
+	int i = 0;
```

Comment:
```suggestion
	int i;
```
Assigned but that value is never used.

Reply:
Done